### PR TITLE
Exception guards for report search fields

### PR DESCRIPTION
**Pull request by SJuliez**
Fixes exceptions thrown in comboXYZ.setSelectedIndex(0) when the search combos in the report display are empty (happened to me when calling up the report at the end of a game).

### CODE REVIEW

File: megamek/src/megamek/client/ui/swing/MiniReportDisplay.java [hunk 1]
```diff
@@ -230,16 +230,17 @@ private void updateEntityChoice() {
...
-        if (comboEntity.getItemCount() == 1) {
-            comboEntity.setEnabled(false);
-        }
         comboEntity.setSelectedItem(lastChoice);
-        if (comboEntity.getSelectedIndex() < 0) {
-            comboEntity.setSelectedIndex(0);
+        if (comboEntity.getItemCount() <= 1) {
+            comboEntity.setEnabled(false);
+        } else {
+            if (comboEntity.getSelectedIndex() < 0) {
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/MiniReportDisplay.java:241]: Any reason to not else if this?

File: megamek/src/megamek/client/ui/swing/MiniReportDisplay.java [hunk 2]
```diff
@@ -252,12 +253,13 @@ private void updateQuickChoice() {
...
-        if (comboQuick.getItemCount() == 1) {
-            comboQuick.setEnabled(false);
-        }
         comboQuick.setSelectedItem(lastChoice);
-        if (comboQuick.getSelectedIndex() < 0) {
-            comboQuick.setSelectedIndex(0);
+        if (comboQuick.getItemCount() <= 1) {
+            comboQuick.setEnabled(false);
+        } else {
+            if (comboQuick.getSelectedIndex() < 0) {
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/MiniReportDisplay.java:260]: Same here

Comment (kuronekochomusuke) [reply megamek/src/megamek/client/ui/swing/MiniReportDisplay.java:260]: think that is for the case when your lastChoice was deleted from the list, to just select the first element

File: megamek/src/megamek/client/ui/swing/MiniReportDisplay.java (hunk 1 above)
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/MiniReportDisplay.java]: Nah :D

**Review by Windchild292 — APPROVED** [review]

**Pull request merged**